Movie_MVVM/Movie_MVVM/MovieListModule/View/MovieListViewController.swift

```diff
@@ -120,63 +155,38 @@ final class MovieListViewController: UIViewController {
         tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
     }
 
-    private func getToDiscriptionVC(path: IndexPath) {
-        let descriptionVS = DescriptionViewController()
-        guard let id = movies?[path.row].id else { return }
-        descriptionVS
-            .urlString = "\(Constants.movieLink)\(id)\(Constants.apiKeyLink)\(Constants.languageLink)"
-        navigationController?.pushViewController(descriptionVS, animated: true)
+    private func createActivityViewAnchor() {
+        NSLayoutConstraint.activate([
+            activityIndicatorView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            activityIndicatorView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
     }
 
     @objc private func segmentControlAction() {
-        switch segmentControl.selectedSegmentIndex {
-        case 0:
-            NetworkManager.fetchGenericData(url: Constants.popularUrl) { [weak self] (movies: MoviesResult) in
-                self?.movies = movies.results
-                DispatchQueue.main.async {
-                    self?.tableView.reloadData()
-                    let indexPath = IndexPath(row: 0, section: 0)
-                    self?.tableView.scrollToRow(at: indexPath, at: .top, animated: true)
-                }
-            }
-        case 1:
-            NetworkManager.fetchGenericData(url: Constants.topUrl) { [weak self] (movies: MoviesResult) in
-                self?.movies = movies.results
-                DispatchQueue.main.async {
-                    self?.tableView.reloadData()
-                    let indexPath = IndexPath(row: 0, section: 0)
-                    self?.tableView.scrollToRow(at: indexPath, at: .top, animated: true)
-                }
-            }
-        case 2:
-            NetworkManager.fetchGenericData(url: Constants.upCommingUrl) { [weak self] (movies: MoviesResult) in
-                self?.movies = movies.results
-                DispatchQueue.main.async {
-                    self?.tableView.reloadData()
-                    let indexPath = IndexPath(row: 0, section: 0)
-                    self?.tableView.scrollToRow(at: indexPath, at: .top, animated: true)
-                }
-            }
-        default:
-            break
-        }
+        movieListViewModel?.segmentControlAction(index: segmentControl.selectedSegmentIndex)
     }
 }
 
+// MARK: - UITableViewDataSource, UITableViewDelegate
```

Comment:
```suggestion
// MARK: - UITableViewDataSource, UITableViewDelegate
```
Нам вроде как говорили разносить по разным экстеншенам...